src/Microsoft.Unity.Analyzers/ImproperMenuItemMethod.cs

```diff
@@ -0,0 +1,102 @@
+/*--------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *-------------------------------------------------------------------------------------------*/
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Unity.Analyzers.Resources;
+
+namespace Microsoft.Unity.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class ImproperMenuItemMethodAnalyzer : DiagnosticAnalyzer
+	{
+		internal static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			id: "UNT0020",
+			title: Strings.ImproperMenuItemMethodDiagnosticTitle,
+			messageFormat: Strings.ImproperMenuItemMethodDiagnosticMessageFormat,
+			category: DiagnosticCategory.Correctness,
+			defaultSeverity: DiagnosticSeverity.Info,
+			isEnabledByDefault: true,
+			description: Strings.ImproperMenuItemMethodDiagnosticDescription);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.RegisterSyntaxNodeAction(AnalyzeMethodDeclaration, SyntaxKind.MethodDeclaration);
+		}
+		private static void AnalyzeMethodDeclaration(SyntaxNodeAnalysisContext context)
+		{
+
+			if (!(context.Node is MethodDeclarationSyntax))
+				return;
+			
+			var method = (MethodDeclarationSyntax)context.Node;
+			var declaredSymbol = context.SemanticModel.GetDeclaredSymbol(method);
+
+			if (!(declaredSymbol is IMethodSymbol methodSymbol))
+				return;
+
+			if (!declaredSymbol.GetAttributes().Any(a => a.AttributeClass.Matches(typeof(UnityEditor.MenuItem))))
+				return;
+
+			if (declaredSymbol.IsStatic)
+				return;
+
+			context.ReportDiagnostic(Diagnostic.Create(Rule, method.GetLocation(), method));
+		}
+	}
+
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class ImproperMenuItemMethodCodeFix : CodeFixProvider
+	{
+		public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(ImproperMenuItemMethodAnalyzer.Rule.Id);
+
+		public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+			var declaration = root.FindNode(context.Span) as MethodDeclarationSyntax;
```

Comment:
root can be `null`, and we can use pattern matching instead of using `as`

```suggestion
			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);

			if (!(root?.FindNode(context.Span) is MethodDeclarationSyntax declaration))
				return;			
```